website/source/docs/internals/providers/resource-data.html.md

```diff
@@ -0,0 +1,180 @@
+`GetChange` is commonly used in cases where the API requires you to explicitly
+remove and add items—for example, when there's a change in tags and the API
+only offers `AddTag` & `RemoveTag` methods without a way to just change all
+tags at once. In that case, you need to know what the previous tag value was,
+so it can be removed.
+
+## Setting State
+
+At the end of your `Create`, `Get`, `Update`, and `Delete` functions, whatever
+is in your `ResourceData` object is set as state. To manipulate this, the `Set`
+method is provided. It takes the key of a property, just like `Get`, as an
+argument, and the value to set it to.
+
+It's important that all your properties get set using the `Set` method, as
+otherwise Terraform will be unable to perform some of its important functions,
+but will give the appearance of operating normally. For example, changes to the
+config file will be detected, but changes made outside Terraform will be
+silently ignored. This breaks Terraform's promise of reflecting your
+infrastructure as code, so it's important that the current state of the
+infrastructure get persisted using the `Set` method.
+
+When using the `Set` method, you shouldn't dereference pointer values, as doing
+so incorrectly (for example, when they're set to a nil value) can crash
+Terraform. The `Set` method is capable of safely derefencing on its own, so the
+safest course of action is to just pass it the pointer.
+
+## Partial Updates
+
+Some resources and data sources cannot be managed using only a single API call;
+for example, a resource may need its permissions set or a compute instance may
+need a disk attached to it, even though the instance and disk are modeled as a
+single resource.
+
+In most cases, this works fine. Even in the case of failures, a the refresh
+part of the lifecycle fixes this. But in some cases (like when refresh is
+disabled), it's helpful to be able to gracefully recover from a failure. For
+these cases, the `*ResourceData` type has
+[`SetPartial`](https://godoc.org/github.com/hashicorp/terraform/helper/schema#ResourceData.SetPartial)
+and
+[`Partial`](https://godoc.org/github.com/hashicorp/terraform/helper/schema#ResourceData.Partial)
+methods. Before we get into how they change Terraform's behaviour, let's talk
+about how Terraform usually works, again.
+
+Usually, when a resource is created or updated, Terraform
+[builds](#what-resourcedata-abstracts) the `*ResourceData` from the current
+state, the config file, and the diff being applied. It then calls your `Create`
+or `Update` function (as appropriate) and any calls to the `*ResourceData`'s
+`Set` method get layered on top of the `*ResourceData`. After _successful_
+completion of the function, the `*ResourceData` gets stored as the new state.
+Should the function return an error, the `*ResourceData` gets discarded and is
+_not_ set as the state.
+
+The `Partial` method accepts a boolean designating whether partial state mode
```

Comment:
Again, I really do not think this guide is useful without code samples. You're basically describing the method signature in a wall of text that could easily be replaced by a code sample of 5-6 lines with a comment.